ttyplot.c

```diff
@@ -610,10 +612,24 @@ int main(int argc, char *argv[]) {
             if (tty >= select_nfds)
                 select_nfds = tty + 1;
         }
-        struct timespec timeout = { .tv_sec = 0, .tv_nsec = 500e6 };  // 500 milliseconds for refreshing the clock
-        const int select_ret = pselect_without_signal_starvation(select_nfds, &read_fds, NULL, NULL, &timeout, &empty_sigset);
 
+        // Determine how long we can allow `select` to sleep.  The rules are:
+        // 1. If we find the system clock has already advanced on seconds level,
+        //    there should be no delay, we should get the clock redrawn asap.
+        // 2. Otherwise wait (more or less) exactly until the system clock
+        //    has advanced, only then redraw the clock.
         gettimeofday(&now, NULL);
+        int timeout_nanos = 0;
+        if (now.tv_sec == prev_paint_at_seconds) {
+            const suseconds_t microseconds_until_full_second = 1000 * 1000 - now.tv_usec;
+            timeout_nanos = 1000 * microseconds_until_full_second;
+        }
+        struct timespec timeout = {
+            .tv_sec = 0,
+            .tv_nsec = timeout_nanos
```

Comment:
This could set `.tv_nsec` to 1e9, which should be avoided.

Reply:
@edgar-bonet I don't see how, please help me see it.

Reply:
If `now.tv_usec` is zero, then `timeout_nanos` ends up being 1e9.

Reply:
@edgar-bonet my bad, classic case of "too obvious to see" :facepalm: Pushed.